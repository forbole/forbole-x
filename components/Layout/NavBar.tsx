import { Box, IconButton } from '@material-ui/core'
import React from 'react'
import Link from 'next/link'
import useStyles from './styles'
// import NotiIcon from '../../assets/images/icons/icon_notification_71.svg'
import SettingsIcon from '../../assets/images/icons/icon_settings.svg'
// import CurrencyMenuButton from './CurrencyMenuButton'
// import LangMenuButton from './LangMenuButton'
import ThemeModeButton from './ThemeModeButton'
import useIconProps from '../../misc/useIconProps'

const NavBar: React.FC<{ HeaderLeftComponent?: React.ReactNode; menuWidth: number }> = ({
  HeaderLeftComponent,
  menuWidth,
}) => {
  const iconProps = useIconProps(3)
  const classes = useStyles()

  return (
    <Box className={classes.navBar}>
      {HeaderLeftComponent ? (
        <Box className={classes.navbarLeftComponentContainer} flex={1} ml={menuWidth}>
          {HeaderLeftComponent}
        </Box>
      ) : null}
      {/* <CurrencyMenuButton />
      <LangMenuButton />
      <IconButton className={classes.navBarButton}>
        <NotiIcon {...iconProps} />
      </IconButton> */}
      <ThemeModeButton />
      <Link href="/settings" passHref>
        <IconButton className={classes.navBarButton}>
          <SettingsIcon {...iconProps} />
        </IconButton>
      </Link>
    </Box>
  )
}

export default React.memo(NavBar)
