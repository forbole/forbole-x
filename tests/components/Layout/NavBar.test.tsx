import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from '../../../components/Layout/NavBar';

jest.mock('../../../components/Layout/CurrencyMenuButton', () => 'CurrencyMenuButton');
jest.mock('../../../components/Layout/LangMenuButton', () => 'CurrencyMenuButton');
jest.mock('../../../components/Layout/ThemeModeButton', () => 'CurrencyMenuButton');

describe('component: Layout - NavBar', () => {
  it('renders correctly', () => {
    const component = renderer.create(<NavBar />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
