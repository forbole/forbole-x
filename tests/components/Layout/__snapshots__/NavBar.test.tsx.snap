// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Layout - NavBar renders correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-8 HookGlobalStyles-navBar-5"
>
  <CurrencyMenuButton />
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-navBarButton-6"
    href="/settings"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_settings"
        fill="#e0e0e0"
        height={24}
        width={24}
      />
    </span>
  </a>
</div>
`;
